test/models/institution_test.rb

```diff
@@ -0,0 +1,124 @@
+    end
+  end
+
+  test "same currency pair cannot be created for the same institution" do
+    assert_difference 'ExchangeRate.count' do
+      @institution.banko.add_rate('USD', 'CAD', 0.9)
+    end
+    assert_no_difference 'ExchangeRate.count' do
+      @institution.banko.add_rate('USD', 'CAD', 0.9)
+    end
+  end
+
+  test "same currency pair can be created for a different institution" do
+    assert_difference 'ExchangeRate.count' do
+      @institution.banko.add_rate('USD', 'CAD', 0.9)
+    end
+    
+    other_institution = Institution.create(name: 'Test Institution', fee: 0)
+    assert_difference 'ExchangeRate.count' do
+      other_institution.banko.add_rate('USD', 'CAD', 0.9)
+    end
+  end
+
+  #
+  # describe get_rate
+  #
+  test "can get_rate" do
+    new_rate = 0.9
+    @institution.banko.add_rate('USD', 'CAD', new_rate)
+
+    assert_equal new_rate, @institution.banko.get_rate('USD', 'CAD')
+  end
+
+  test "returns nil when currency pair does not exist" do
+    assert_nil @institution.banko.get_rate('USD', 'CLP')
+  end
+
+  test "same currency pair for the same institution returns the same rate" do
+    @institution.banko.add_rate('USD', 'CAD', 0.9)
+
+    assert_equal @institution.banko.get_rate('USD', 'CAD'), @institution.banko.get_rate('USD', 'CAD')
+  end
+
+  test "same currency pair for the different institution returns the different rate" do
+    @institution.banko.add_rate('USD', 'CAD', 0.9)
+    first_ins_rate = @institution.banko.get_rate('USD', 'CAD')
+
+    other_institution = Institution.create(name: 'Test Institution', fee: 0)
+    other_institution.banko.add_rate('USD', 'CAD', 0.8)
+    second_ins_rate = other_institution.banko.get_rate('USD', 'CAD')
+
+    assert_not_equal first_ins_rate, second_ins_rate
+  end
+
+  #
+  # describe get_amount
+  #
+  test "can convert amount" do
+    @institution.banko.add_rate('USD', 'CLP', 750)
+
+    converted_amount = @institution.get_amount('USD', 'CLP', 3)
+    assert_equal 750 * 3, converted_amount.amount
+  end
+
+  test "returns nil amount when currency pair does not exist" do
+    assert_nil @institution.get_amount('USD', 'CLP', 10)
+  end
+
+  test "same currency pair for the same institution returns the same amount" do
+    @institution.banko.add_rate('USD', 'CAD', 0.9)
+
+    amount_one = @institution.get_amount('USD', 'CAD', 10)
+    amount_two = @institution.get_amount('USD', 'CAD', 10)
+    assert_equal amount_one, amount_two
+  end
+
+  test "same currency pair for the different institution returns the different amount" do
+    @institution.banko.add_rate('USD', 'CAD', 0.9)
+    first_ins_amount = @institution.get_amount('USD', 'CAD', 10)
+
+    other_institution = Institution.create(name: 'Test Institution', fee: 0)
+    other_institution.banko.add_rate('USD', 'CAD', 0.8)
+    second_ins_amount = other_institution.get_amount('USD', 'CAD', 10)
+
+    assert_not_equal first_ins_amount, second_ins_amount
+  end
+end
```

Comment:
tests for input types?
from_iso_code should be string,
to_iso_code  should be string,
from_amount  should be int